src/buffer/mod.rs

```diff
@@ -580,9 +584,7 @@ impl TextBuffer {
         self.redo_stack.clear();
         self.last_history_type = HistoryType::Other;
         self.cursor = Default::default();
-        self.cursor_for_rendering = None;
         self.set_selection(None);
-        self.search = None;
```

Comment:
These two removed lines are old, from the initial import. I've since added dirty tracking in the ICU code which makes resetting `self.search` superfluous.